components/mpas-ocean/src/shared/mpas_ocn_eddy_parameterization_helpers.F

```diff
@@ -0,0 +1,282 @@
+
+     type (mpas_pool_type), pointer, intent(in) :: statePool
+
+     integer :: iEdge, nEdges, refIndex, cell1, cell2, nCells, k, iCell
+     real(kind=RKIND) :: dDenThres, den_ref_lev
+     real(kind=RKIND),dimension(:), allocatable :: depth
+     integer, dimension(:), pointer :: landIceMask
+     real (kind=RKIND), dimension(:,:), pointer :: layerThickness
+     logical :: found_den_mld
+     real (kind=RKIND) :: dV, dVp1, refDepth, coeffs(2), mldTemp
+
+     call mpas_pool_get_array(statePool, 'layerThickness', layerThickness, 1)
+     nCells = nCellsAll
+
+     allocate(depth(nVertLevels))
+     refDepth = config_mld_reference_depth
+
+    !$omp parallel
+    !$omp do schedule(runtime) &
+    !$omp private(depth, refIndex, coeffs, found_den_mld, k, den_ref_lev, dVp1, dV, mldTemp)
+    do iCell = 1,nCells
+
+       found_den_mld = .false.
+
+       depth(1) = 0.5_RKIND*layerThickness(1,iCell)
+       do k=2,maxLevelCell(iCell)
+          depth(k) = depth(k-1) + 0.5_RKIND*(layerThickness(k-1,iCell) + &
+                                             layerThickness(k  ,iCell))
+       end do
+
+       do k=1, maxLevelCell(iCell)-1
+          if(depth(k+1) >= refDepth) then
+             coeffs(2) = (potentialDensity(k+1,iCell) - potentialDensity(k,iCell)) &
+                          / (depth(k+1) - depth(k))
+             coeffs(1) = potentialDensity(k,iCell) - coeffs(2)*depth(k)
+
+             den_ref_lev = coeffs(2)*refDepth + coeffs(1)
+
+             refIndex=k
+             exit
+          end if
+       end do
+
+       do k=refIndex,maxLevelCell(iCell)-1
+
+          if(.not. found_den_mld .and. (potentialDensity(k+1,iCell) - den_ref_lev) .ge. &
+                   config_mixedLayerDepths_crit_dens_threshold) then
+             dVp1 = (potentialDensity(k+1,iCell) - den_ref_lev)
+             dV   = (potentialDensity(k  ,iCell) - den_ref_lev)
+
+             coeffs(2) = (depth(k+1) - depth(k)) / (dVp1 - dV)
+             coeffs(1) = depth(k) - coeffs(2)*dV
+
+             mldTemp = coeffs(2)*config_mixedLayerDepths_crit_dens_threshold + coeffs(1)
+             mldTemp=min(mldTemp,depth(k+1)) !make sure MLD isn't deeper than zMid(k+1)
+             dThreshMLD(iCell)=abs(max(mldTemp,depth(k))) !MLD should be deeper than zMid(k)
+             indMLD(iCell) = k+1
+             found_den_mld = .true.
+             exit
+          end if
+       end do
+
+       ! if the mixed layer depth is not found, it is set to the depth of the bottom most level
+       if(.not. found_den_mld) then
+          dThreshMLD(iCell) = depth(maxLevelCell(iCell))
+          indMLD(iCell) = maxLevelCell(iCell)
+       end if
+    end do !iCell
+    !$omp end do
+    !$omp end parallel
+
+    nEdges = nEdgesHalo(1)
+    !$omp parallel
+    !$omp do schedule(runtime) &
+    !$omp private(cell1,cell2)
+    do iEdge=1,nEdgesAll
+       cell1 = cellsOnEdge(1,iEdge)
+       cell2 = cellsOnEdge(2,iEdge)
+       indMLDedge(iEdge) = min(indMLD(cell1),indMLD(cell2))
```

Comment:
@philipwjones I think I could move this to a calculation in the submesocale eddy calculation to remove this from ocn_diagnostics_variables if you prefer that approach.